timeseries/src/autogluon/timeseries/trainer/abstract_trainer.py

```diff
@@ -878,15 +862,174 @@ def score(
+        pred_time_dict_total = defaultdict(float)
+        for model_name in pred_time_dict_marginal.keys():
+            for base_model in self.get_minimum_model_set(model_name):
+                if pred_time_dict_marginal[base_model] is not None:
+                    pred_time_dict_total[model_name] += pred_time_dict_marginal[base_model]
+        return dict(pred_time_dict_total)
+
+    @property
+    def _cached_predictions_filename(self) -> Path:
+        return Path(self.path) / "cached_predictions.pkl"
+
+    @staticmethod
+    def _compute_dataset_hash(
+        data: TimeSeriesDataFrame, known_covariates: Optional[TimeSeriesDataFrame] = None
+    ) -> str:
+        """Compute a unique string that identifies the time series dataset."""
+        from hashlib import md5
+
+        def hash_df(df: Optional[pd.DataFrame]) -> str:
+            if df is not None:
+                df = pd.DataFrame(data)
+                df.sort_index(inplace=True, axis=0)
+                df.reset_index(inplace=True)
+                df.sort_index(inplace=True, axis=1)
+                hashable_object = pd.util.hash_pandas_object(df).values
+            else:
+                hashable_object = "0".encode("utf-8")
+            return md5(hashable_object).hexdigest()
+
+        combined_hash = hash_df(data) + hash_df(known_covariates) + hash_df(data.static_features)
+        return md5(combined_hash.encode("utf-8")).hexdigest()
+
+    def _get_cached_pred_dicts(self, dataset_hash: str) -> Tuple[Dict[str, TimeSeriesDataFrame], Dict[str, float]]:
+        """Load cached predictions for given dataset_hash from disk, if possible. Otherwise returns empty dicts."""
+        if self._cached_predictions_filename.exists():
+            cached_predictions = load_pkl.load(str(self._cached_predictions_filename))
+            if dataset_hash in cached_predictions:
+                model_pred_dict = cached_predictions[dataset_hash]["model_pred_dict"]
+                pred_time_dict = cached_predictions[dataset_hash]["pred_time_dict"]
+                if model_pred_dict.keys() == pred_time_dict.keys():
+                    logger.debug(f"Loaded cached predictions for models {list(model_pred_dict.keys())}")
+                    return model_pred_dict, pred_time_dict
+                else:
+                    logger.warning(f"Found corrupted cached predictions in {self._cached_predictions_filename}")
+        logger.debug(f"Found no cached predictions")
+        return {}, {}
+
+    def _save_cached_pred_dicts(
+        self, dataset_hash: str, model_pred_dict: Dict[str, TimeSeriesDataFrame], pred_time_dict: Dict[str, float]
+    ) -> None:
+        if self._cached_predictions_filename.exists():
+            logger.debug(f"Extending existing cached predictions")
+            cached_predictions = load_pkl.load(str(self._cached_predictions_filename))
+        else:
+            cached_predictions = {}
```

Comment:
Theoretically this file could get cumbersome if we were to save many caches. Another option would be to save files `cache_N.pkl`, with a dictionary of `dataset_hash -> filename`. Don't worry about actually implementing this though, it is only relevant if such a scenario is realistic.